script.js

```diff
@@ -0,0 +1,123 @@
+	// Script
+    document.addEventListener('DOMContentLoaded', () => {
+        // Get the wrapper.
+            const block = document.querySelector('.q-and-a');
+    
+        // Get the video wrapper.
+            const video = block.querySelector('.q-and-a__video');
+            let player = null;
+    
+            // Initiate the video player.
+            if (video) {
+                player = new Plyr(video.querySelector('video'), {
+                    active: true,
+                    hideControls: true,
+                });
+            }
+    
+        /* 
+         * Get all the buttons. Maybe ideally this would create an array with the      * index and the data-video attribute.
+         */
+            const buttonWrap = block.querySelector('.q-and-a__video-buttons');
+            const buttons = buttonWrap ? Array.from(buttonWrap.querySelectorAll('.progress-btn')) : null;
+        let currentIndex;
+        
+        /*
+         * Events to trigger while playing.
+         * Grab the player duration - unavailable until the video plays. 
+         */
+        player.on('playing', () => {
+          duration = player.duration;
+    
+          /*
+           * While playing, get the current time / duration. This gives a
+           * percentage complete - ex. .45
+           * Multiply that by 100 to get the percentage in a form we can 
+           * attach to a clip-path. 
+           * Set that number to the css variable. 
+           * Do this as in interval to reduce the number of calls - maybe this
+           * could be better.
+           */
+          btnProgress = setInterval(() => {
+            currentTime = player.currentTime;
+            const float = (currentTime / duration).toFixed(2);
+            const progress = `${float * 100}%`;
+            buttons[currentIndex].style.setProperty('--progress-width', progress);
+          }, 300);
+        });
+    
+        /*
+         * Video end events. Clear the interval. Reset times. 
+         * Get the current button index - maybe if my button array was set up
+         * better, I could already have this instead of looping again. 
+         * Get the next video. If we're on the last video, get the 1st video.
+         */
+        player.on('ended', () => {
```

Comment:
nitpick (non-blocking): When binding a callback to an event, a best practice is to move the function to a variable. Why this is recommended is 2-fold. 

1. When we name a callback, and pass the variable name to the binding, we can then clean it up if we want to remove the callback further into the logic. 
2. When we name a callback, we can use it directly in tests if we write them. When we create a callback such as this, it becomes much more difficult to test the inner functionality.

eg.
```
const endedCallback = () => { ... };

player.on('ended', endedCallback);
```

This can then allow us to use an `off` if the API allows.

```
player.off('ended', endedCallback);
```

Of course this depends on the API that player uses. But this is especially useful when dealing with `addEventListener` and `removeEventListener`